### PR TITLE
Development

### PR DESCRIPTION
**Pull request by diego-ons**
# Description

Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style and contribution guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

## Issue this pull request resolves

Changes that address updates from the Uganda and Malawi trip, as well as some extra modifications.
Merging changes ahead of change of repository.

### CODE REVIEW

**Comment by diego-ons** [comment]
There is some associated "tech-debt" as a result of this. Mainly, there are missing tests for a couple of the new funcitons.
These are captured in issue #57 .

File: R/create_table_5_1.R
Comment (jeroenminderman) [inline R/create_table_5_1.R:1]: Using example, function gives an error for me currently:

```
death_data <- construct_sample_death_data()
death_data <- construct_year(death_data, date_col = "death1a", year_col = "dodyr")
t1 <- create_t5.1(data = death_data, tablename = "Table_5_1")
```

image

File: R/create_table_5_6_&_5_7.R
Comment (jeroenminderman) [inline R/create_table_5_6_&_5_7.R:1]: Fails if using following example: 

```
death_data <- construct_sample_death_data()
death_data <- construct_year(death_data, date_col = "death1a", year_col = "dodyr")
death_data <- construct_age(death_data, dob_col = "death2a", age_col = "death2b", end_date_col = "death1a")
death_data <- construct_age_group(death_data, age_col = "death2b", new_col = "death2b_grp")
table_5_6 <- create_t5.6_and_t5.7(death_data, ru_filter = "urban", age_var = "death2b_grp", data_year = 2022)
```

image

File: R/create_table_5_11.R [hunk 1]
```diff
@@ -15,20 +27,30 @@
...
 #' @examples
-#' t5.11 <- create_t5.11(bth_data, date_var = dobyr, tablename = "Table_5_11")
+#' death_data <- construct_sample_death_data()
+#' death_data <- construct_year(death_data, date_col = "death1a", year_col = "dodyr")
+#' table_5_11 <- create_t5.11(death_data, neonatal_filter = NULL, date_var= "dodyr")
 #'
-create_t5.11 <- function(data, date_var, num_yrs = 5, tablename = NA){
-  latest_year <- data %>% pull(!!sym(date_var)) %>% max(na.rm = TRUE)
-  years <- generate_year_sequence(latest_year = latest_year, num_yrs = num_yrs)
+create_t5.11 <- function(data, foetal_filter = "Yes", num_yrs = 5,
```
Comment (jeroenminderman) [inline R/create_table_5_11.R:34]: Example uses "neonatal_filter"

File: R/create_table_5_12.R
Comment (jeroenminderman) [inline R/create_table_5_12.R:1]: Function fails currently using following example. Likely data missing in example?
```
death_data <- construct_sample_death_data()
death_data <- construct_year(death_data, date_col = "death1a", year_col = "dodyr")
table_5_12 <- create_t5.12(death_data)
```

image

**Review by jeroenminderman — COMMENTED** [review]
I've checked all Section 5 functions using examples given in docstrings and using `construct_sample_death_data()` only. 
I can't get 5.1, 5.6-5.7 or 5.12 to work immediately - examples used and output included in review comments.
Have not had chance to troubleshoot further, appologies

**Review by jeroenminderman — APPROVED** [review]
Added minor fixes as per latest commits.
As a note only for future ref - currently 5.1, 5.6-5.7 and 5.12 create table functions do not appear to produce outputs using docstring examples
@diego-ons

**Pull request merged**